adapters/resetdigital/resetdigital.go

```diff
@@ -0,0 +1,284 @@
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+
+		requestBody, err := json.Marshal(splittedRequestData)
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+
+		requests = append(requests, &adapters.RequestData{
+			Method:  "POST",
+			Uri:     a.endpointUri,
+			Body:    requestBody,
+			Headers: getHeaders(requestData),
+		})
+	}
+
+	return requests, errors
+}
+
+func processDataFromRequest(requestData *openrtb2.BidRequest, imp openrtb2.Imp, bidType openrtb_ext.BidType) (resetDigitalRequest, error) {
+	var resetDigitalRequestData resetDigitalRequest
+
+	// Check if requestData.Site is not nil before accessing its fields
+	if requestData.Site != nil {
+		resetDigitalRequestData.Site.Domain = requestData.Site.Domain
+		resetDigitalRequestData.Site.Referrer = requestData.Site.Page
+	}
+
+	resetDigitalRequestData.Imps = append(resetDigitalRequestData.Imps, resetDigitalRequestImps{})
+	resetDigitalRequestData.Imps[0].BidID = requestData.ID
+	resetDigitalRequestData.Imps[0].ImpID = imp.ID
+
+	if bidType == openrtb_ext.BidTypeBanner {
+		resetDigitalRequestData.Imps[0].MediaTypes.Banner.Sizes = append(resetDigitalRequestData.Imps[0].MediaTypes.Banner.Sizes, []int64{imp.Banner.Format[0].W, imp.Banner.Format[0].H})
+	}
+	if bidType == openrtb_ext.BidTypeVideo {
+		resetDigitalRequestData.Imps[0].MediaTypes.Video.Sizes = append(resetDigitalRequestData.Imps[0].MediaTypes.Banner.Sizes, []int64{*imp.Video.W, *imp.Video.H})
+	}
+
+	var extData = make(map[string]interface{})
+	err := json.Unmarshal(imp.Ext, &extData)
+	if err != nil {
+		return resetDigitalRequestData, err
+	}
+
+	resetDigitalRequestData.Imps[0].ZoneID.PlacementID = extData["bidder"].(map[string]interface{})["placement_id"].(string)
+	if resetDigitalRequestData.Imps[0].ZoneID.PlacementID == "test" {
+		resetDigitalRequestData.Imps[0].ForceBid = true
+	}
+
+	return resetDigitalRequestData, nil
+}
+
+func (a *adapter) MakeBids(internalRequest *openrtb2.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if response.StatusCode == http.StatusNoContent || response.StatusCode == http.StatusBadRequest || response.StatusCode != http.StatusOK {
+		return nil, nil
+	}
+
+	var bidResp map[string]interface{}
+	if err := json.Unmarshal(response.Body, &bidResp); err != nil {
+		return nil, []error{err}
+	}
+
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(maxBids)
+	jsonData := bidResp["bids"].([]interface{})
+
+	for _, bidData := range jsonData {
+		bidMap := bidData.(map[string]interface{})
+		bid := getBidFromResponse(bidMap)
+		if bid == nil {
+			continue
+		}
+
+		bidTypes, err := getBidTypes(internalRequest.Imp[0])
+		if err != nil {
+			return nil, []error{err}
+		}
+
+		for _, bidType := range bidTypes {
+			bidResponse.Bids = append(bidResponse.Bids, &adapters.TypedBid{
+				Bid:     bid,
+				BidType: bidType,
+			})
+		}
+	}
```

Comment:
Avoid using maps and type casting values to `[]interface{}, map[string]interface{}` Instead make use of structs. Refer following as example

https://github.com/prebid/prebid-server/blob/dd07431ef54c38a2617c04e76b3e7bda9c21f569/adapters/appnexus/appnexus.go#L167-L205